tests/bluetooth/iso/bsim/main.c

```diff
@@ -0,0 +1,452 @@
+#define CENTRAL_SOURCE	1
+#define PERIPHERAL_SINK 2
+
+#define COLOR_RED   "\x1B[0;31m"
+#define COLOR_GREEN "\x1B[0;32m"
+#define COLOR_BLUE  "\x1B[0;34m"
+#define COLOR_RESET "\x1b[0m"
+
+static uint8_t role;
+extern enum bst_result_t bst_result;
+static int argc_copy;
+static char argv_copy[MAX_ARGS][MAX_ARG_LEN];
+
+enum transport {
+	ACL,
+	ISO,
+};
+
+/* This code is executed by the test framework and not Zephyr */
+static void test_args(int argc, char *argv[])
+{
+	argc_copy = argc;
+
+	for (int i = 0; i < argc; i++) {
+		if (i > MAX_ARGS) {
+			TEST_FAIL("ENOMEM");
+		}
+		if (strlen(argv[i]) > MAX_ARG_LEN) {
+			TEST_FAIL("ENOMEM");
+		}
+		memcpy(argv_copy[i], argv[i], strlen(argv[i]) + 1);
+	}
+}
+
+struct test_report {
+	uint32_t last_count;
+	uint32_t counts_fail;
+	uint32_t counts_success;
+	double fail_percentage;
+	bool passed;
+};
+
+/* Report structure for ACL and ISO results */
+static struct test_report report[2];
+static uint8_t fail_limit_percent[] = {0, 0};
+static uint32_t packets_limit_min[] = {UINT32_MAX, UINT32_MAX};
+
+static void result_check(uint32_t last_count, uint32_t counts_fail, uint32_t counts_success,
+			 enum transport trans)
+{
+
+	if (role == CENTRAL_SOURCE) {
+		return;
+	}
+
+	report[trans].last_count = last_count;
+	report[trans].counts_fail = counts_fail;
+	report[trans].counts_success = counts_success;
+
+	if (counts_success == 0) {
+		report[trans].fail_percentage = 100;
+	} else {
+		report[trans].fail_percentage =
+			(counts_fail * 100) / (double)(counts_success + counts_fail);
+	}
+
+	/* Ensure enough packets have been transferred */
+	if (report[ACL].counts_success < packets_limit_min[ACL] ||
+	    report[ISO].counts_success < packets_limit_min[ISO]) {
+		return;
+	}
+
+	if (report[ACL].fail_percentage > fail_limit_percent[ACL]) {
+		TEST_FAIL("FAIL percentage %05.2f > %d", report[ACL].fail_percentage,
+			  fail_limit_percent[ACL]);
+	} else {
+		report[ACL].passed = true;
+	}
+
+	if (report[ISO].fail_percentage > fail_limit_percent[ISO]) {
+		TEST_FAIL("FAIL percentage %05.2f > %d", report[ISO].fail_percentage,
+			  fail_limit_percent[ISO]);
+	} else {
+		report[ISO].passed = true;
+	}
+
+	if (report[ACL].passed && report[ISO].passed) {
+		bst_result = 0; /* Set as passed */
+	}
+}
+
+static void sim_acl_recv_main_cb(uint32_t last_count, uint32_t counts_fail, uint32_t counts_success)
+{
+	result_check(last_count, counts_fail, counts_success, ACL);
+}
+
+static void sim_iso_recv_main_cb(uint32_t last_count, uint32_t counts_fail, uint32_t counts_success)
+{
+	result_check(last_count, counts_fail, counts_success, ISO);
+}
+
+static bool is_number(const char *str)
+{
+	if (*str == '\0') {
+		return 0;
+	}
+	uint8_t len = strlen(str);
+	/* Iterate through each character of the string */
+	for (uint8_t i = 0; i < len; i++) {
+		/* Check if the character is not a digit */
+		if (!isdigit((int)*str)) {
+			return false;
+		}
+		str++;
+	}
+
+	return true;
+}
+
+static int cmd_send(size_t argc, ...)
```

Comment:
```suggestion
static int cmd_shell_send(size_t argc, ...)
```

Or something like that. `cmd_send` is very generic, and suggest to include `shell` in the name to indicate that it's sending a command to the shell